onnxruntime/contrib_ops/cuda/bert/skip_layer_norm.cc

```diff
@@ -60,22 +60,31 @@ Status SkipLayerNorm<T, Simplified>::ComputeInternal(OpKernelContext* ctx) const
   // of the input and skip tensors
   Tensor* skip_input_bias_add_output = ctx->Output(3, input->Shape());
 
-  if (input->Shape() != skip->Shape()) {
+  const auto& input_dims = input->Shape().GetDims();
+  size_t input_dims_size = input_dims.size();
+  const auto& skip_dims = skip->Shape().GetDims();
+  size_t skip_dims_size = skip_dims.size();
+  
+  if (skip_dims_size != 3 && skip_dims_size != 2) {
     return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
-                           "skip is expected to have same shape as input");
-  }
-
-  if (input->Shape().Size() == 0) {
-    return Status::OK();
+                           "skip is expected to have 3 or 2 dimensions, got ", skip_dims_size);
   }
 
-  const auto& input_dims = input->Shape().GetDims();
-  size_t input_dims_size = input_dims.size();
   if (input_dims_size != 3 && input_dims_size != 2) {
     return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
                            "input is expected to have 3 or 2 dimensions, got ", input_dims_size);
   }
 
+  if ((input->Shape() != skip->Shape()) && ((skip_dims[0] != 1 || skip_dims_size != 2) && input_dims_size != 3)) {
```

Comment:
The check on CPU and CUDA is the same. We can move the login to a helper function. See example in https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/contrib_ops/cpu/bert/multihead_attention_helper.h.